modules/servicetemplate/src/main/java/org/eclipse/xpanse/modules/servicetemplate/ServiceTemplateManage.java

```diff
@@ -193,14 +202,24 @@ public ServiceTemplateEntity registerServiceTemplateByUrl(String oclLocation) th
      * @return Returns service template DB entity.
      */
 
-    public ServiceTemplateEntity getServiceTemplateDetails(String id) {
+    public ServiceTemplateEntity getServiceTemplateDetails(String id, boolean checkNamespace) {
         UUID uuid = UUID.fromString(id);
         ServiceTemplateEntity serviceTemplateEntity = storage.getServiceTemplateById(uuid);
         if (Objects.isNull(serviceTemplateEntity)) {
             String errMsg = String.format("Service template with id %s not found.", id);
             log.error(errMsg);
             throw new ServiceTemplateNotRegistered(errMsg);
         }
+        if (checkNamespace) {
+            CurrentUserInfo currentUserInfo = identityProviderManager.getCurrentUserInfo();
+            if (Objects.nonNull(currentUserInfo)) {
+                if (!StringUtils.equals(serviceTemplateEntity.getNamespace(),
+                        currentUserInfo.getNamespace())) {
+                    throw new AccessDeniedException("No right to view details of service template "
```

Comment:
No permissions to view details of services templates belonging to other namespaces.